packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx

```diff
@@ -303,25 +305,53 @@ const OrderDetailsTable: React.FC<OrderDetailsProps> = ({ patientUuid, showAddBu
     [orderTypes, t],
   );
 
+  const handleDateFilterChange = ([startDate, endDate]) => {
+    if (startDate) {
+      setSelectedFromDate(startDate.toISOString());
+      if (selectedToDate && startDate && selectedToDate < startDate) {
+        setSelectedToDate(startDate.toISOString());
+      }
+    }
```

Comment:
```suggestion
    if (startDate) {
    	const isoStartDate = startDate.toISOString();
      setSelectedFromDate(isoStartDate);
      if (selectedToDate && selectedToDate < startDate) {
        setSelectedToDate(isoStartDate);
      }
    }
```

The outer `startDate` check in the if condition should be sufficient.